Assets/Scripts/InventoryAndItemsScripts/Items/Interactables/UIInteractable.cs

```diff
@@ -39,29 +45,43 @@ public void EnableInteractableCanvas()
             this.interactableOverlay?.EnableCanvas();
         }
 
-        public void Interact()
+        public virtual void Interact()
         {
             this.isActive = !this.isActive;
```

Comment:
this is convenient, but toggling states like this tends to lead to a lot of bugs later down the line. I'd make separate StartInteract and StopInteract methods